The checks parameter in OIDCConfig causes a type error

### Environment

```
  System:
    OS: macOS 12.6
    CPU: (8) arm64 Apple M1
    Memory: 103.48 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 20.9.0 - ~/.asdf/installs/nodejs/20.9.0/bin/node
    npm: 10.1.0 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 119.0.6045.105
    Safari: 15.6.1


  npmPackages:
    next-auth: ^5.0.0-beta.3 => 5.0.0-beta.3
```

### Reproduction URL

https://github.com/nbifrye/76956150-50fe-4232-b78f-dba38e1d6801

### Describe the issue

Specifying the checks option for custom OAuth provider causes a type error.

```
% npx tsc index.ts --noEmit                                           
index.ts:11:12 - error TS2322: Type 'string' is not assignable to type 'never'.

11   checks: ["none"],
```

The type of `checks` is as follows, which means `("pkce" | "state" | "none")[] & "nonce"[]`. The only value that satisfies this is `[]`.

https://github.com/nextauthjs/next-auth/blob/1349ed33da3280162bd8156a80e0a4659cb873dd/packages/core/src/providers/oauth.ts#L242

The expected type of checks looks like `("pkce" | "state" | "none" | nonce")[]`. How about the following code:

```
checks?: Array<(Exclude<OAuth2Config<Profile>["checks"], undefined>)[number] | "nonce">
```


### How to reproduce

```
git clone https://github.com/nbifrye/76956150-50fe-4232-b78f-dba38e1d6801.git
cd 76956150-50fe-4232-b78f-dba38e1d6801
npm install
npx tsc --noEmit
```


### Expected behavior

`checks` such as `["none"]`, `["state", "nonce"]`, etc. do not cause a type error.

--- a/packages/core/src/providers/oauth.ts
+++ b/packages/core/src/providers/oauth.ts
@@ -239,7 +239,7 @@ export interface OAuth2Config<Profile>
 export interface OIDCConfig<Profile>
   extends Omit<OAuth2Config<Profile>, "type" | "checks"> {
   type: "oidc"
-  checks?: OAuth2Config<Profile>["checks"] & Array<"nonce">
+  checks?: Array<(Exclude<OAuth2Config<Profile>["checks"], undefined>)[number] | "nonce">
 }
 
 export type OAuthConfig<Profile> = OIDCConfig<Profile> | OAuth2Config<Profile>

